modules/demo/ssr/point-cloud/render/render.js

```diff
@@ -0,0 +1,155 @@
+const copyFramebuffer = require('./copy')();
+
+class Renderer {
+  constructor() {
+    const onAnimationFrameRequested = immediateAnimationFrame(this);
+    const jsdom                     = new RapidsJSDOM({module, onAnimationFrameRequested});
+
+    const {deck, render} = jsdom.window.evalFn(makeDeck);
+
+    this.deck    = deck;
+    this.jsdom   = jsdom;
+    this._render = render;
+  }
+  async render(props = {}, graph = {}, state = {}, events = [], frame = 0) {
+    const window = this.jsdom.window;
+
+    props && this.deck.setProps(props);
+    state?.deck && this.deck.restore(state.deck);
+    state?.window && Object.assign(window, state.window);
+
+    (events || []).forEach((event) => window.dispatchEvent(event));
+
+    await this._render();
+
+    return {
+      frame: copyFramebuffer(this.deck.animationLoop, frame),
+      state: {
+        deck: this.deck.serialize(),
+        window: {
+          x: window.x,
+          y: window.y,
+          title: window.title,
+          width: window.width,
+          height: window.height,
+          cursor: window.cursor,
+          mouseX: window.mouseX,
+          mouseY: window.mouseY,
+          buttons: window.buttons,
+          scrollX: window.scrollX,
+          scrollY: window.scrollY,
+          modifiers: window.modifiers,
+          mouseInWindow: window.mouseInWindow,
+        },
+      }
+    };
+  }
+}
+
+module.exports.Renderer = Renderer;
+
+function immediateAnimationFrame(renderer) {
+  let request  = null;
+  let flushing = false;
+  const flush = () => {
+    flushing = true;
+    while (request && request.active) {
+      const f = request.flush;
+      request = null;
+      f();
+    }
+    flushing = false;
+  };
+  return (r) => {
+    if (flushing) { return request = r; }
+    if (renderer?.deck?.animationLoop?._initialized) {  //
+      return flush(request = r);
+    }
+    if (!request && (request = r)) { setImmediate(flush); }
+  };
+}
+
+function makeDeck() {
+  const {log: deckLog} = require('@deck.gl/core');
+  deckLog.level        = 0;
+  deckLog.enable(false);
+
+  const {OrbitView, COORDINATE_SYSTEM, LinearInterpolator} = require('@deck.gl/core');
+  const {PointCloudLayer}                                  = require('@deck.gl/layers');
+  const {TextLayer}                                        = require('@deck.gl/layers');
+  const {DeckSSR}                                          = require('@rapidsai/deck.gl');
+  const {LASLoader}                                        = require('@loaders.gl/las');
+  const {registerLoaders}                                  = require('@loaders.gl/core');
+
+  registerLoaders(LASLoader);
+
+  // Data source: kaarta.com
+  const LAZ_SAMPLE =
+    'https://raw.githubusercontent.com/uber-common/deck.gl-data/master/examples/point-cloud-laz/indoor.0.1.laz';
+
+  const transitionInterpolator = new LinearInterpolator(['rotationOrbit']);
+
+  const makeLayers = (deck, graph = null) => {
```

Comment:
I think renaming `graph` to `pointCloud` might lead to less confusion?
```suggestion
  const makeLayers = (deck, pointCloud = null) => {
```